lib/collection/url.js

```diff
@@ -243,7 +243,7 @@ _.assign(Url.prototype, /** @lends Url.prototype */ {
             segmentArray = (options && options.unresolved) ? this.path : _.transform(this.path, function (res, segment) {
                 var variable;
 
-                if (_.startsWith(segment, PATH_VARIABLE_IDENTIFIER)) {
+                if (_.startsWith(segment, PATH_VARIABLE_IDENTIFIER) && segment.length > 1) {
```

Comment:
@codenirvana - i know you have explained this to me once... but I keep forgetting the nature of the change added by segment.length > 1 -> can you document it inline so that in future it is more clear why we did this.